module itself (changed)

kamaev opened the pull request.

Commit:
Create moira_trigger.py

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
rewritten module

Commit:
rewritten README.md

Commit:
Update README.md

Commit:
fix README.md

Commit:
prettify code and impove doc

Commit:
module output improvements

Commit:
code improvements

Commit:
auth keys added

Commit:
update README.md

beevee commented:
Уже можно смотреть, или ты еще доделываешь?

kamaev commented:
Да, можно смотреть. Последние коммиты - это я уже сам к себе придираюсь как могу, но думаю уже в любом случае если что то ещё менять, то лучше уже на основе ревью :)

beevee commented on moira_trigger.py:
```diff
@@ -0,0 +1,338 @@
+
+    def parameters_change(self, moira_trigger, trigger):
+
+        parameters_to_change = {}
```
По-моему, эта переменная нигде не используется.

kamaev replied:
Убрал переменную. Спасибо! Как то совсем не обратил внимания на эту наркоманию

beevee commented on moira_trigger.py:
```diff
@@ -0,0 +1,338 @@
+                self.results[trigger['name']] = {'new trigger created': moira_trigger.id}
+                
+            self.parameters_change(moira_trigger, trigger)
+
```
Ты прогонял PEP8 или какие-то линтеры? Тут, кажется, две строки должно быть между верхнеуровневыми функциями и классами. Если не прогонял, то прогони и исправь замечания.

kamaev replied:
Прогнал через pep8, pyflakes, flake8, pylint. Внёс правки за исключением конфликтующих с требованиями Ansible для импортирования модулей после RETURNS (https://docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html)

beevee commented on moira_trigger.py:
```diff
@@ -0,0 +1,338 @@
+        for component in components:
+            try:
+                moira.trigger.trigger_client.get('/'+component)
+            except:
```
Тут теряется информация о том, какой эксепшен вылетел. Можно ли использовать эту информацию в сообщении об ошибке для пользователя? Грубо говоря, чтобы он отличал no route to host от ошибки 500 со стороны апи. Даже если сейчас из клиента всегда вылетают одинаковые ошибки, которые невозможно отличить - в будущем это может измениться.

kamaev replied:
Добавил exception_handler() и запись в failed через него

beevee commented on moira_trigger.py:
```diff
@@ -0,0 +1,338 @@
+
+        for moira_trigger in all_triggers:
+            if moira_trigger.name == trigger['name']:
+                trigger_id = moira_trigger.id
```
Тут сразу можно return делать, не просматривать остальные триггеры.

kamaev replied:
Исправил

beevee commented on moira_trigger.py:
```diff
@@ -0,0 +1,338 @@
+
+        if state == 'absent':
+            
+            if current_id:
```
Лучше явно проверять is not None, мне кажется.

kamaev replied:
Исправил

beevee commented on moira_trigger.py:
```diff
@@ -0,0 +1,338 @@
+
+        elif state == 'present':
+            
+            if current_id:
```
Тут тоже.

kamaev replied:
Исправил

beevee commented on moira_trigger.py:
```diff
@@ -0,0 +1,338 @@
+    if not moira_ansible.api_check():
+        module.fail_json(msg=moira_ansible.failed)
+
+    moira_ansible.triggers(trigger, module.params['state'])
```
А тут ничего не может сломаться? Кажется, никак не проверяется, удачно ли обновился триггер.

kamaev replied:
Добавил функцию 'trigger_update_check' для проверки, и изменил зависимости успешного/неуспешного выхода из модуля

beevee commented on moira_trigger.py:
```diff
@@ -0,0 +1,338 @@
+        module.fail_json(msg=moira_ansible.failed)
+
+    moira_ansible.triggers(trigger, module.params['state'])
+    moira_ansible.tag_cleanup()
```
Тут тоже.

kamaev replied:
Добавил warning для неудачного завершения 'tag_cleanup', т.к. ф-ция выполняется уже после попытки изменить состояния триггера, при неудачном выполнении которого в любом случае должен вылететь failed, в случае если состояние триггера было удачно изменено, но тэги не потерлись также будет warning (с сообщением о том, что теги могут быть потерты на следующем запуске плейбука)

beevee commented on moira_trigger.py:
```diff
@@ -0,0 +1,338 @@
+
+        moira_trigger.update()
+
+    def triggers(self, trigger, state):
```
Неудачное имя для функции. Нет никакого глагола.

kamaev replied:
Переименовал функцию в 'trigger_customize'

Commit:
fixes

module documentation:
- docstrings added
- module doc changed
functions:
- function 'triggers' renamed to 'trigger_customize
- function 'parameters_change' renamed to 'trigger_update'
- added function 'trigger_update_check'
- added function 'exception_handler'
checks:
- added warning for tag_cleanup

and other improvements

Commit:
update README.md

Commit:
removed obvious statement

beevee commented:
Мне норм. Ты как-то тестировал этот модуль? Работает хоть? :)

kamaev commented:
Да, прогонял на moira-vagrant-dev.
Сами тесты пока переписываю с учетом того, чтобы можно было их и для Travis использовать и в качестве поддержки модулем Dry Runs.
